src/kudu/consensus/consensus_peers.cc

```diff
@@ -629,7 +673,20 @@ void RpcPeerProxy::UpdateAsync(const ConsensusRequestPB* request,
                                rpc::RpcController* controller,
                                const rpc::ResponseCallback& callback) {
   controller->set_timeout(MonoDelta::FromMilliseconds(FLAGS_consensus_rpc_timeout_ms));
-  consensus_proxy_->UpdateConsensusAsync(*request, response, controller, callback);
+
+  boost::optional<std::string> rpc_token = request->has_raft_rpc_token()
+                                               ? request->raft_rpc_token()
+                                               : boost::optional<std::string>();
+  bool enforce = controller->enforce_rpc_token();
```

Comment:
What is the reason for needing extra copy in rpc_controller, and not using ShouldEnforceRaftRpcToken() in raft_consensus or in consensus_queue? Is it because consensus_peer does not have reference to them?

Having three copies of the same value makes update harder.

Reply:
consensus_peers is a separate module from raft_consensus and consensus_queue. It shouldn't know about any of those as part of the abstraction contract.

rpc_controller is initialized per request, so there's no need for any action when the value is updated. The next request will simply pick it up.

Reply:
I agree with the general sentiment here brought up by Chi. We are threading enforce_rpc_token into all parts of the system which don't need to have any threading required. This is not a knob that is going to be required after rollout phase and too low level for threading overhead. We should try to use a global, or a single call somewhere to whether to enforce or not.